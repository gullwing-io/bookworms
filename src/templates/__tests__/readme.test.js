import {
  bookmark,
  bookmarkFolder,
  convertNumberIntoHeader,
  bookmarkLink,
  addTitle,
  addDescription,
  addBookwormsDescription,
} from "../readme";

// I don't like using snapshots for these tests but because of the HTML formatting its easier
describe("Templates - Readme", () => {
  describe("bookmark", () => {
    test("should return structure for Readme bookmarks", () => {
      expect(
        bookmark(
          "26/09/2021 - 6pm",
          "Bookmarks label",
          "Bookmarks description",
          "_Children markdown_"
        )
      ).toMatchSnapshot();
    });
    test("should return structure for Readme bookmarks with blank label", () => {
      expect(
        bookmark(
          "26/09/2021 - 6pm",
          "",
          "Bookmarks description",
          "_Children markdown_"
        )
      ).toMatchSnapshot();
    });
  });

  describe("bookmarkFolder", () => {
    test("should return description, header and children", () => {
      expect(
        bookmarkFolder(
          0,
          "Section label",
          "Section description",
          "_Children markdown_"
        )
      ).toMatchSnapshot();
    });
  });

  describe("convertNumberIntoHeader", () => {
    test("should return h1 mardown", () => {
      expect(convertNumberIntoHeader(0, "h1")).toEqual("# h1");
    });
    test("should return h2 mardown", () => {
      expect(convertNumberIntoHeader(1, "h2")).toEqual("## h2");
    });
    test("should return h5 mardown", () => {
      expect(convertNumberIntoHeader(4, "h5")).toEqual("##### h5");
    });
  });

  describe("bookmarkLink", () => {
    test("should return description and link", () => {
      expect(
        bookmarkLink(
          "26/09/2021 - 6pm",
          "Link label",
          "Link description",
          "https://www.testlink.com"
        )
      ).toMatchSnapshot();
    });
  });

  describe("addTitle", () => {
    test("should return title comment", () => {
      expect(addTitle("hello world")).toEqual("# hello world");
    });
    test("should return empty string", () => {
      expect(addTitle("")).toEqual("");
    });
  });

  describe("addDescription", () => {
    test("should return description comment", () => {
      expect(addDescription("hello world")).toEqual("- hello world");
    });
    test("should return empty string", () => {
      expect(addDescription("")).toEqual("");
    });
  });

  describe("addBookwormsDescription", () => {
    test("should return description for file comment", () => {
      expect(addBookwormsDescription(1633027394454)).toEqual(
        expect.stringContaining("_These bookmarks were last updated on"),
        expect.stringContaining(
          "using [Bookworms](https://github.com/thearegee/bookworms)_"
        )
      );
    });
  });
});
