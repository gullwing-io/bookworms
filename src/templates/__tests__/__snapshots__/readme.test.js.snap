// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Templates - Readme bookmark should return structure for Readme bookmarks 1`] = `
"# Bookmarks label

Bookmarks description
_Children markdown_
_These bookmarks were last updated on Invalid Date using [Bookworms](https://github.com/thearegee/bookworms)_"
`;

exports[`Templates - Readme bookmark should return structure for Readme bookmarks with blank label 1`] = `
"

Bookmarks description
_Children markdown_
_These bookmarks were last updated on Invalid Date using [Bookworms](https://github.com/thearegee/bookworms)_"
`;

exports[`Templates - Readme bookmarkFolder should return description, header and children 1`] = `
"

# Section label

Section description
_Children markdown_"
`;

exports[`Templates - Readme bookmarkLink should return description and link 1`] = `
"
* [26/09/2021 - 6pm](Link description) - Link label
"
`;
